### PR TITLE
REST API - VM Analysis

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: cfme/tests/infrastructure/test_vm_analysis.py -k REST -v --long-running }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 8a03b06494a6ea98ad8cef2d096aa400af5a0999:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/infrastructure/test_vm_analysis.py [hunk 1]
```diff
@@ -302,35 +302,53 @@ def is_vm_analysis_finished(vm_name):
     return vm_analysis_finished is not None
 
 
-def _scan_test(provider_crud, vm, os, fs_type, soft_assert):
+def _scan_test(provider_crud, vm, os, fs_type, soft_assert, rest, rest_api):
```
Comment (seandst) [inline cfme/tests/infrastructure/test_vm_analysis.py]: This is overloaded, and should be broken out into two functions.

File: cfme/tests/infrastructure/test_vm_analysis.py [hunk 2]
```diff
@@ -390,6 +408,8 @@ def _scan_test(provider_crud, vm, os, fs_type, soft_assert):
         soft_assert("os-linux_fedora.png" in details_os_icon)
         soft_assert("linux_fedora" in quadicon_os_icon)
 
+pytestmark = [pytest.mark.parametrize("rest", ["using_ui", "rest_detail", "rest_collection"])]
```
Comment (seandst) [inline cfme/tests/infrastructure/test_vm_analysis.py]: Using parametrize to create new test cases is a little dirty, and it'll probably need to be undone (and be easier to read and debug) when _scan_test is broken up.

Comment (mfalesni) [inline cfme/tests/infrastructure/test_vm_analysis.py]: Ok, will extract that out to a separate class.

**Comment by dajoRH** [comment]
Lint report for commit 1c207d088a0642406fb55f1e6566423e614b2a2b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6144619ff94ee2d29e7e7f3f5f59ccb0c2c6e4f9:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**